src/coreclr/src/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -471,11 +475,19 @@ protected ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type,
                 {                    
                     Debug.Assert(fieldType.IsPrimitive || fieldType.IsPointer || fieldType.IsFunctionPointer || fieldType.IsEnum);
 
-                    var fieldSizeAndAlignment = ComputeFieldSizeAndAlignment(fieldType, packingSize);
                     instanceNonGCPointerFieldsCount[CalculateLog2(fieldSizeAndAlignment.Size.AsInt)]++;
                 }
             }
 
+            largestAlignmentRequired = type.Context.Target.GetObjectAlignment(largestAlignmentRequired);
+
+            if (largestAlignmentRequired.IsIndeterminate)
```

Comment:
This check is not necessary. Calling AsInt is going to throw if the value is indeterminate. This might actually be reachable at some point. Once it is we will deal with it.